pkg/client/unversioned/request.go

```diff
@@ -587,6 +590,24 @@ func (r *Request) Watch() (watch.Interface, error) {
 	return watch.NewStreamWatcher(watchjson.NewDecoder(resp.Body, r.codec)), nil
 }
 
+// updateUrlMetrics is a convenience function for pushing metrics.
+// It also handles corner cases for incomplete/invalid request data.
+func updateUrlMetrics(req *Request, resp *http.Response, err error) {
+	url := "none"
+	// In unit tests, weird urls can occur.
```

Comment:
Agree, but am assuming that in general gaurding against nils is a good thing... probably ill just remove the comment, since in and of itself, checking nil is still a good thing to do.